platforms/core-execution/daemon-server-worker/src/main/java/org/gradle/workers/internal/WorkerProcessIsolationProblemsServiceProvider.java

```diff
@@ -0,0 +1,188 @@
+
+        ClassLoader parent = WorkerAction.class.getClassLoader();
+        FilteringClassLoader.Spec filterSpec = new FilteringClassLoader.Spec();
+        FilteringClassLoader modelClassLoader = new FilteringClassLoader(parent, filterSpec);
+
+        return new PayloadSerializer(
+            new WellKnownClassLoaderRegistry(
+                new DefaultPayloadClassLoaderRegistry(
+                    classLoaderCache,
+                    new ModelClassLoaderFactory(modelClassLoader))));
+    }
+
+    @Provides
+    PatternSpecFactory createPatternSpecFactory(ListenerManager listenerManager) {
+        PatternSpecFactory patternSpecFactory = PatternSpecFactory.INSTANCE;
+        listenerManager.addListener(patternSpecFactory);
+        return patternSpecFactory;
+    }
+
+    @Provides
+    DirectoryFileTreeFactory createDirectoryFileTreeFactory(Factory<PatternSet> patternSetFactory, FileSystem fileSystem) {
+        return new DefaultDirectoryFileTreeFactory(patternSetFactory, fileSystem);
+    }
+
+    @Provides
+    Factory<PatternSet> createPatternSetFactory(final PatternSpecFactory patternSpecFactory) {
+        return PatternSets.getPatternSetFactory(patternSpecFactory);
+    }
+
+    @Provides
+    ClassLoaderHierarchyHasher createClassLoaderHierarchyHasher() {
+        return classLoader -> {
+            throw new UnsupportedOperationException();
+        };
+    }
+
+    @Provides
+    IsolatableSerializerRegistry createIsolatableSerializerRegistry(
+        ManagedFactoryRegistry managedFactoryRegistry,
+        ClassLoaderHierarchyHasher classLoaderHierarchyHasher
+    ) {
+        return new IsolatableSerializerRegistry(classLoaderHierarchyHasher, managedFactoryRegistry);
+    }
+
+    @Provides
+    IsolatableFactory createIsolatableFactory(
+        ManagedFactoryRegistry managedFactoryRegistry,
+        ClassLoaderHierarchyHasher classLoaderHierarchyHasher
+    ) {
+        return new DefaultIsolatableFactory(classLoaderHierarchyHasher, managedFactoryRegistry);
+    }
+
+    @Provides
+    PropertyHost createPropertyHost() {
+        return PropertyHost.NO_OP;
+    }
+
+    @Provides
+    CrossBuildInMemoryCacheFactory createCrossBuildInMemoryCacheFactory(ListenerManager listenerManager) {
+        return new DefaultCrossBuildInMemoryCacheFactory(listenerManager);
+    }
+
+    @Provides
+    NamedObjectInstantiator createNamedObjectInstantiator(CrossBuildInMemoryCacheFactory cacheFactory) {
+        return new NamedObjectInstantiator(cacheFactory);
+    }
+
+    @Provides
+    ManagedFactoryRegistry createManagedFactoryRegistry(
+        NamedObjectInstantiator namedObjectInstantiator,
+        InstantiatorFactory instantiatorFactory,
+        PropertyFactory propertyFactory,
+        FileCollectionFactory fileCollectionFactory,
+        FileFactory fileFactory,
+        FilePropertyFactory filePropertyFactory
+    ) {
+        return new DefaultManagedFactoryRegistry().withFactories(
```

Comment:
Don't we already create one of these in `WorkerSharedGlobalScopeServices#createManagedFactoryRegistry`? It seems we are defining a lot of the same factories too between these. 

If we really need to create another, I think we should be using the constructor that accepts a parent factory registry.